src/middlewared/middlewared/plugins/docker/update.py

```diff
@@ -66,6 +66,15 @@ async def validate_data(self, old_config, config, schema='docker_update'):
                 'System is not licensed to use Applications'
             )
 
+        if config['pool'] and (await self.middleware.call('system.security.config'))['enable_gpos_stig']:
+            verrors.add(
+                f'{schema}.pool',
+                'Apps cannot be enabled under General Purpose OS STIG compatibility mode.'
+            )
+
+        # No point in contining if we're in STIG or not licensed
```

Comment:
There's no way to enable this in GPOS compatibility mode. Even FULL_ADMIN doesn't have DOCKER_WRITE privilege. I don't understand the check.